samples/bluetooth/st_ble_sensor/src/button_svc.c

```diff
@@ -0,0 +1,100 @@
+/** @file
+ *  @brief Button Service sample
+ */
+
+/*
+ * Copyright (c) 2019 Marcio Montenegro <mtuxpe@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+#include <zephyr.h>
+#include <drivers/gpio.h>
+#include <logging/log.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/gatt.h>
+#include "button_svc.h"
+
+LOG_MODULE_REGISTER(button_svc);
+
+/* GPIO button; change this to use another GPIO port */
+#ifndef DT_ALIAS_SW0_GPIOS_CONTROLLER
+#ifdef SW0_GPIO_NAME
+#define DT_ALIAS_SW0_GPIOS_CONTROLLER SW0_GPIO_NAME
+#else
+#error SW0_GPIO_NAME or DT_ALIAS_SW0_GPIOS_CONTROLLER needs to be set in board.h
+#endif
+#endif
+#define PORT    DT_ALIAS_SW0_GPIOS_CONTROLLER
+
+/* change this to use another GPIO pin */
+#ifdef DT_ALIAS_SW0_GPIOS_PIN
+#define PIN     DT_ALIAS_SW0_GPIOS_PIN
+#else
+#error DT_ALIAS_SW0_GPIOS_PIN needs to be set in board.h
+#endif
+
+extern struct bt_conn *p_conn;
+extern struct bt_gatt_service_static stsensor_svc[];
+extern volatile u8_t notify_enable;
+
+static struct device *button_dev;
+static struct gpio_callback gpio_cb;
+u16_t but_val;
+
+void button_pressed(struct device *gpiob, struct gpio_callback *cb,
+		    u32_t pins)
+{
+	int err;
+
+	LOG_INF("Button SW1 pressed");
+
+	if (p_conn) {
+
+		if (notify_enable == true) {
+
+			err = bt_gatt_notify(NULL, &stsensor_svc->attrs[2],
+				(const void *)&but_val, sizeof(but_val));
+			if (err) {
+				LOG_ERR("Notify err");
```

Comment:
How about logging the exact value of err with %d?

Reply:
done